desktop/cmp/contextmenu/ContextMenu.js

```diff
@@ -63,14 +65,15 @@ export class ContextMenu extends Component {
             return true;
         }).map(item => {
             if (item === '-') return menuDivider();
-            if (isReactElement(item))  return item;
-
+            if (isReactElement(item)) {
+                return menuItem({text: item});
+            }
+            
             const items = item.items ? this.parseMenuItems(item.items) : null;
-
             return menuItem({
                 text: item.text,
                 icon: item.icon,
-                onClick: () => start(item.action),    // do async to allow menu to close
+                onClick: item.action ? () => start(item.action) : null,    // do async to allow menu to close
```

Comment:
Nice approach here - good to know!  Always dislike it when the menu keeps floating around for a tick while the action commences behind it.